src/validation/fields.ts

```diff
@@ -87,63 +87,57 @@ export const redirectTo = Joi.string()
       }
     }
 
-    // * We allow any sub-path of the client url
-    // * With optional hash and query params
-    if (
-      new RegExp(`^${ENV.AUTH_CLIENT_URL}(/.*)?([?].*)?([#].*)?$`).test(value)
-    ) {
-      return value;
-    }
+    const regexpContainsPort = new RegExp(`https?://[^/]+(:\\d+)(.*)`);
+    const regexpAddPort = new RegExp(`(https?://[^/]+)(.*)`);
+
+    const matches: string[] = [];
+
+    for (let url of [
+      ...ENV.AUTH_ACCESS_CONTROL_ALLOWED_REDIRECT_URLS,
+      ENV.AUTH_CLIENT_URL,
+    ]) {
+      switch (true) {
+        case url.endsWith('/**'):
+          break;
+        case url.endsWith('/*'):
+          url += '*';
```

Comment:
will this modify the original string?

Reply:
this will mutate the `url` but here we're just looping through a copy of `AUTH_ACCESS_CONTROL_ALLOWED_REDIRECT_URLS` and so the array of the allowed redirect urls won't be mutated